[frontend-api] Products can now be loaded via frontend API (#1471)
shopsys/shopsys: docs/frontend-api/introduction-to-frontend-api.md

```

You can override resolvers like any other Symfony service.

> ### Resolvers Maps

Reviewer: ```suggestion
### Resolver Maps
```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6

> If we map GrapQl objects to entities, it will may happen that automatic transformation is not possible.

Reviewer: ```suggestion
If we map GraphQl objects to entities, it may happen that automatic transformation is not possible.
```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6

> For example, this can happen when we want to extend an entity with only an item for graphql (such as link to  a product).

Reviewer: ```suggestion
This can happen when we want to use getter for some entity attribute and such getter requires parameter.
```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6

> Or if the entity contains an item whose getter requires a required parameter.

Reviewer: ```suggestion

```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6

For this transformation we can use a `ResolverMap` object .
`ResolverMap` is a Symfoy service that implements `Overblog\GraphQLBundle\Resolver\ResolverMapInterface`.
ResolverMap can be created as a child of `Overblog\GraphQLBundle\Resolver\ResolverMap` class too and overload the `map` method.

> Example of `ResolveMap`:

Reviewer: ```suggestion
Example of `ResolverMap`:
```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6

```php
class ProductResolverMap extends ResolverMap
{
    /**
     * @return array
     */
    protected function map(): array
    {
        return [
            'Product' => [
                'shortDescription' => function (Product $product) {
                    return $product->getShortDescription($this->domain->getId());
                },
                'link' => function (Product $product) {
                    return $this->getProductLink($product);
                },
            ],
        ];
    }
}
```

> You  can register `ResolveMap` in `app/config/packages/shopsys_frontend_api.yml`:

Reviewer: ```suggestion
You  can register `ResolverMap` in `app/config/packages/shopsys_frontend_api.yml`:
```

Author: Fixed https://github.com/shopsys/shopsys/pull/1471/commits/6d8726cb09dea957f5b76d7b82744e4c3603a6b6
